databuilder/extractor/bigquery_usage_extractor.py

```diff
@@ -0,0 +1,110 @@
+from collections import namedtuple
+from datetime import date, timedelta
+import logging
+
+from googleapiclient.discovery import build
+from oauth2client.client import GoogleCredentials
+
+from databuilder.extractor.base_extractor import Extractor
+
+TableColumnUsageTuple = namedtuple('TableColumnUsageTuple', ['database', 'cluster', 'schema',
+                                                             'table', 'column', 'email'])
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BigQueryTableUsageExtractor(Extractor):
+    """
+    An aggregate extractor for bigquery table usage. This class takes the data from
+    the stackdriver logging API by filtering on timestamp, bigquery_resource and looking
+    for referencedTables in the response.
+    """
+    TIMESTAMP_KEY = 'timestamp'
+    PROJECT_ID_KEY = 'projectid'
+    DEFAULT_PAGE_SIZE = 300
+    PAGE_SIZE_KEY = 'pagesize'
+
+    def init(self, conf):
+        credentials = GoogleCredentials.get_application_default()
+        self.logging_service = build('logging', 'v2', credentials=credentials)
+
+        # type: (ConfigTree) -> None
+        self.timestamp = conf.get_string(
+            BigQueryTableUsageExtractor.TIMESTAMP_KEY,
+            (date.today() - timedelta(days=1)).strftime('%Y-%m-%dT00:00:00Z'))
+        self.projectid = conf.get_string(BigQueryTableUsageExtractor.PROJECT_ID_KEY)
+        self.pagesize = conf.get_int(
+            BigQueryTableUsageExtractor.PAGE_SIZE_KEY,
+            BigQueryTableUsageExtractor.DEFAULT_PAGE_SIZE)
+        self.table_usage_counts = {}
+        self._count_usage()
+        self.iter = iter(self.table_usage_counts)
+
+    def _count_usage(self):
+        count = 0
+        for entry in self._retrieve_records():
+            count += 1
+            if count % self.pagesize == 0:
+                LOGGER.info('Aggregated {} records'.format(count))
+
+            email = entry['protoPayload']['authenticationInfo']['principalEmail']
```

Comment:
@feng-tao @Alagappan : We should probably check for job "success" completion status as well?  At the moment also jobs that completed but failed will be included?

Reply:
@Alagappan may have better idea as he works on BQ at Lyft. But I am fine as first cut to only check jobComplete. @gtoonstra let me know what you think?

Reply:
Good catch @gtoonstra! would be good to keep the usage counts consistent across rest of the codebase